includes/class-woocommerce.php

```diff
@@ -8,6 +8,9 @@
 
 defined( 'ABSPATH' ) || exit;
 
+require_once __DIR__ . '/../src/Theming/ThemeSupport.php';
```

Comment:
It will get already included by the autoloader.

Reply:
The problem is that I'm using the `ThemeSupport` class before plugins are loaded, and if I remove that line I get the following:

> Notice: Automattic\WooCommerce\Theming\ThemeSupport was called <strong>incorrectly</strong>. Not all plugins have loaded yet but we requested the class Automattic\WooCommerce\Theming\ThemeSupport Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version dev-fix/25900.) in /Users/konamiman/localsites/wp-54/app/public/wp-includes/functions.php on line 5167

Reply:
This is a limitation of the Jetpack Autoloader. There is a [PR being worked on](https://github.com/Automattic/jetpack/pull/15106) for a 2.0 of the autoloader that removes this limitation, but for the time being, this is probably the only way we can get around it. Admittedly it entirely defeats the point of the autoloader, but, we can't inject the dependencies otherwise.